File: advent.of.code/2015/cpp/day01/floor.search.cpp
//Reuben Thorpe (2015) 1st December Advent of Code C++11
#include <iostream>
#include <fstream>
#include <sstream>

using namespace std;

int main() {
  int floor = 0, index;
  ifstream input("./input.txt");

  if (input.is_open()) {
    stringstream buffer;
    buffer << input.rdbuf();
    input.close();
    bool flag = false;
    char step;

    //Part 1
    while (buffer >> step) {
      switch(step) {
        case '(': ++floor; break;
        case ')': --floor; break;
      }
      //Part 2
      if (!flag && floor == -1) {
        index = buffer.tellg();
        flag = true;
      }
    }

  }
  cout << "\nPart 1 = " << floor << endl;
  cout << "Part 2 = " << index << "\n\n";
  return 0;

}

File: advent.of.code/2015/python/day01/floor.search.py
# Reuben Thorpe (2015) , 1st December Advent of code


def findFloor():
    data = open('input.txt', 'r').read()
    floor = 0
    flag = False
    for index, i in enumerate(data, start=1):
        if (i == "("):
            floor += 1
        elif (i == ")"):
            floor -= 1

        if (not flag and floor == -1):
            basmentIndex = index
            flag = True

    print("\nPart 1 = ", floor)
    print("Part 2 = ", basmentIndex, "\n")


findFloor()
